src/argowrapper/engine/argo_engine.py

```diff
@@ -486,7 +486,18 @@ def workflow_submission(self, request_body: Dict, auth_header: str):
             ARGO_NAMESPACE, request_body, auth_header, WORKFLOW.GWAS
         )
         workflow_yaml = workflow._to_dict()
+
+        # If billing_id exists for user, add it to workflow label and pod metadata
+        # remove gen3-username from pod metadata
+        if request_body.get("billing_id"):
+            billing_id = request_body.get("billing_id")
+            workflow_yaml["metadata"]["labels"]["billing_id"] = billing_id
+            pod_labels = workflow_yaml["spec"]["podMetadata"]["labels"]
+            pod_labels["billing_id"] = billing_id
```

Comment:
not sure why, but we've been prefixing the labels with `gen3` (maybe there was a reason for that - not to conflict with other labels...not sure).